gimie/sources/common/license.py

```diff
@@ -1,34 +1,50 @@
+import os
 import re
-from spdx_license_list import LICENSES
+from spdx_license_list import LICENSES, License
 from scancode.api import get_licenses
-from typing import List
+from typing import Iterable, List, Optional
 from gimie.io import Resource, iterable_to_stream, RemoteResource
+import tempfile
 
 
-def _get_licenses(temp_file_path: str) -> str:
-    """Takes a file with a license text in it, and matches this using the scancode API to a license to get some possible
-    license matches. The highest match is then returned as a spdx license ID"""
-    license_detections = get_licenses(temp_file_path, include_text=True)[
+def _get_license_url(license_file: Resource) -> Optional[str]:
+    """Takes the path of a text file containing a license text, and matches this
+    using the scancode API to get possible license matches. The best match is
+    then returned as a spdx license URL"""
+    temp_file = tempfile.NamedTemporaryFile(delete=False)
+    temp_file.write(license_file.open().read())
+    temp_file.close()
+
+    license_detections = get_licenses(temp_file.name, include_text=True)[
         "license_detections"
     ]
-    license_id = get_license_with_highest_coverage(license_detections)
-    spdx_license_id = get_spdx_license_id(LICENSES, license_id)
+    license_id = get_license_with_highest_coverage(license_detections)  # type: ignore
+    spdx_license_id = get_spdx_license_id(LICENSES.keys(), license_id)
+    os.unlink(temp_file.name)
```

Comment:
So, reading the docs of os.unlink - this does the same as os.remove. Is there a reason you chose for the (in my opinion) less representative "unlink"? Essentially, we want to remove the temporary file completely, so .remove seems very appropriately worded.

Reply:
Good to know, it was just a habit, but os.remove is clearer indeed 
864d0e4b9d76945ef19702b7483fbac71760d235 switches to it